contributingGuides/philosophies/PAGINATION.md

```diff
@@ -0,0 +1,113 @@
+    - A good starting point for this is to write your query using a [row value comparison](https://www.sqlite.org/rowvalue.html). Then, if necessary, consider creating a sorted `COVERING` index for the row value you're searching over. For example:
+
+        ```sql
+        CREATE TABLE chatMessages (
+            id       INTEGER PRIMARY KEY,     -- unique message ID
+            created  DATETIME NOT NULL,       -- message creation timestamp
+            message  TEXT NOT NULL
+        );
+
+        -- Create a sorted covering index for the cursor (sorting by created, tie-breaking with IDs)
+        CREATE INDEX idx_chatMessages_created_id
+            ON chatMessages (created, id);
+
+        -- Query a single page using your cursor as the "OFFSET"
+        SELECT *
+        FROM chatMessages
+        WHERE (created, id) < ('2025-09-01 10:02:00', 3)
+        ORDER BY created DESC, id DESC
+        LIMIT 50;
+        ```
+
+4. Determine whether your use-case calls for **unidirectional** or **bidirectional** pagination. Generally, **bidirectional** pagination will be useful if:
+
+    - The list live-updates and the user can "jump" to an arbitrary point in the list without loading all the data between the start of the list and that point.
+    - The list doesn't live-update (i.e: if a user scrolls back in the list, and while they're looking at older items, they're not receiving newer items as they come in).
+
+    If you need bidirectional pagination, you'll need to ensure that you craft queries and API endpoints to fetch data from a given cursor in both directions. In the example query above, it could be as simple as switching `<` to `>=` and making the comparator and sort order explicit for each direction.
+
+> [!NOTE]
+> At the time of writing, [RecyclerListView](https://github.com/Flipkart/recyclerlistview) and [FlashList](https://github.com/Shopify/flash-list) _do not_ support bidirectional pagination.
+> React Native's built-in [FlatList](https://reactnative.dev/docs/flatlist) and [legend-list](https://github.com/LegendApp/legend-list) are some examples that _do_ support bidirectional pagination via the `onStartReached` prop.
+
+5. Determine whether it's possible for **gaps** to appear in your data. It's non-trivial to list all the ways gaps can appear in a list, but here are a couple of examples, one real and one contrived:
+    - "Comment linking"
+        1. User has a simple paginated list of integers in ascending order, and it currently contains items 1-50.
+        2. User jumps to the middle of the list (say for simplicity that they are looking at items 15-35)
+        3. While they're looking at the middle of the list, more than 1 page (50 items) of data is added to the front of the list. Let's say these items are items 50-150.
+        4. They jump back to the front of the page, and fetch the page at the front, 100-150.
```

Comment:
```suggestion
        4. They jump to the front of the list and then fetch page 100-150.
```

Reply:
In lieu of a complete explanation of "comment linking" we could also say "They deep link to a comment in the front" or something like that. It might not be clear what it means to "jump".